unstructured/metrics/evaluate.py

```diff
@@ -585,3 +604,95 @@ def filter_metrics(
         raise ValueError("Please provide `export_filename`.")
     else:
         raise ValueError("Return type must be either `dataframe` or `file`.")
+
+
+@dataclass
+class ObjectDetectionMetricsCalculator(BaseMetricsCalculator):
+    """
+    Calculates object detection metrics for each document:
+    - f1 score
+    - precision
+    - recall
+    - average precision (mAP)
+    It also calculates aggregated metrics.
+    """
+
+    def __post_init__(self):
+        super().__post_init__()
+        self._document_paths = [
+            path.relative_to(self.documents_dir)
+            for path in self.documents_dir.rglob("analysis/*/layout_dump/object_detection.json")
+        ]
+
+    @property
+    def supported_metric_names(self):
+        return ["f1_score", "precision", "recall", "m_ap"]
+
+    @property
+    def default_tsv_name(self):
+        return "all-docs-object-detection-metrics.tsv"
+
+    @property
+    def default_agg_tsv_name(self):
+        return "aggregate-object-detection-metrics.tsv"
+
+    def _find_file_in_ground_truth(self, file_stem: str):
+        for path in self._ground_truth_paths:
+            if Path(path.stem).stem == file_stem:
+                return path
+        return None
+
+    def _process_document(self, doc: Path) -> list:
+        od_dump_path = Path(doc)
+        file_stem = od_dump_path.parts[-3]
+
+        src_gt_filename = self._find_file_in_ground_truth(file_stem)
+
+        if src_gt_filename not in self._ground_truth_paths:
+            return None
+
+        doctype = Path(src_gt_filename.stem).suffix[1:]
```

Comment:
What's the reason to use the `.stem` and `Path` over `src_gt_filename.suffix`?
If there's multiple suffixes perhaps `.suffixes` would be more transparent.

Reply:
That needs a comment too :) 
So for ground truth file like:
```
some_document.pdf.json
```
we need to take the middle - `pdf` in this case - as a source file type. This operation supports more complicated cases like:
```
some_document.tar.gz.pdf.json
```
-> `tar.gz.pdf`
(they happen sometimes in datasets)

Reply:
I've added descriptive docstrings and comments here and there - I think it's clear now :)